### ISSUE TITLE
Stitching not available

### ISSUE THREAD

**Issue by tval2** [elevenlabs/elevenlabs-js#64]
I am trying to keep my app low latency and hence am using the streaming api via this node library. I also want to maintain decent flow between my chunks that I am sending in (I am splitting on new sentences) and hence need to stitch them together using the request_ids of previous streams.

However, upon looking into this further it seems that `client.generate` has no way of also returning the request_id as an optional output and therefore I can't even use this feature at all. I was told in the discord to try the HTTP API but the output format of the HTTP API differs from the SDK, so I need to make other changes in my codebase simply because swapped out how I'm calling the API. 

I tried looking in the source code here to see if I could mimic the output of the SDK but it doesn't seem to work. So figured I'd post here asking for a PR where you output the request_id if asked as well.

**Comment by dsinghvi** [comment]
@tval2 have you considered using `textToSpeech.convert` ?

**Comment by tval2** [comment]
> @tval2 have you considered using `textToSpeech.convert` ? 

Maybe i'm missing something but my understanding was that convert only returns the response body, no? At least that's what it shows in the repo.

**Comment by tval2** [comment]
@dsinghvi perhaps more broadly: is there any clean way to use the SDK and pass the previous_request_ids parameter at the same time? The `generate` call allows for it so I still feel like I'm missing something

**Comment by tval2** [comment]
Also on this topic, a contradictory guideline from the API reference:

- In the API [docs](https://elevenlabs.io/docs/api-reference/streaming) it says  `In case both previous_text and previous_request_ids is send, previous_text will be ignored. `

- In this [tutorial](https://elevenlabs.io/docs/api-reference/how-to-use-request-stitching) it says `best possible results are achieved when conditioning both on text and past generations so lets combine the two by providing previous_text, next_text and previous_request_ids in one request`

Which one should we follow? Should I send previous_text or no?

**Comment by dsinghvi** [comment]
@tval2 you should be able to use `textToSpeech.convertAsStream` for the streaming api. Additionally that method supports both parameters that you mention such as [previous_request_ids](https://github.com/elevenlabs/elevenlabs-js/blob/58beeb10866574d797c3adae20001077b7793c4f/src/api/resources/textToSpeech/client/requests/StreamTextToSpeechRequest.ts#L48-L50)

**Comment by ofekrom** [comment]
I am also encountering the same issue reported in the original message

**Comment by ceifa** [comment]
> @tval2 you should be able to use `textToSpeech.convertAsStream` for the streaming api. Additionally that method supports both parameters that you mention such as [previous_request_ids](https://github.com/elevenlabs/elevenlabs-js/blob/58beeb10866574d797c3adae20001077b7793c4f/src/api/resources/textToSpeech/client/requests/StreamTextToSpeechRequest.ts#L48-L50)

It's not possible to get the request id from textToSpeech.convertAsStream

**Comment by shreecodes** [comment]
> @tval2 have you considered using `textToSpeech.convert` ?

@dsinghvi Neither `textToSpeech.convert` nor `textToSpeech.convertAsStream`, nor `generate` return the `request_id`, which is required to pass into subsequent requests.

> Also on this topic, a contradictory guideline from the API reference:
> ...
> Which one should we follow? Should I send previous_text or no?
@tval2 I noticed the same thing 😅

**Comment by PaulAsjes** [comment]
Hi all, we're tracking this and are looking into a fix where we always return request IDs in SDK responses.

**Comment by garkpit** [comment]
I guess using a direct call using fetch or axios would allow me to get the request-id header.  Do you see any reason not to go this route?

**Comment by PaulAsjes** [comment]
@garkpit Yes you could go down that route if you need the request IDs immediately. Only downside is that you don't get the type checking that you'd expect when using an SDK.

**Comment by vyconm** [comment]
2.0+ resolves this! thanks!

**Comment by PaulAsjes** [comment]
@vyconm You beat me to it :) official docs are incoming but you can see an example of how to do request stitching here: https://github.com/elevenlabs/elevenlabs-js/wiki/v2-upgrade-guide#request-stitching
